bundles/org.openhab.binding.miio/src/main/java/org/openhab/binding/miio/internal/handler/MiIoBasicHandler.java

```diff
@@ -394,36 +413,39 @@ private void updatePropsFromJsonObject(MiIoSendCommand response) {
-                val = transformed;
+        if (basicChannel == null) {
+            logger.debug("Channel not found for {}", param);
+            return;
+        }
+        final String transformation = basicChannel.getTransfortmation();
+        if (transformation != null) {
+            JsonElement transformed = Conversions.execute(transformation, val);
+            logger.debug("Transformed with '{}': {} {} -> {} ", transformation, basicChannel.getFriendlyName(), val,
+                    transformed);
+            val = transformed;
+        }
+        try {
+            if (basicChannel.getType().equals("Number")) {
+                updateState(basicChannel.getChannel(), new DecimalType(val.getAsBigDecimal()));
             }
-            try {
-                if (basicChannel.getType().equals("Number")) {
-                    updateState(basicChannel.getChannel(), new DecimalType(val.getAsBigDecimal()));
-                }
-                if (basicChannel.getType().equals("String")) {
-                    updateState(basicChannel.getChannel(), new StringType(val.getAsString()));
-                }
-                if (basicChannel.getType().equals("Switch")) {
-                    updateState(basicChannel.getChannel(), val.getAsString().toLowerCase().equals("on")
-                            || val.getAsString().toLowerCase().equals("true") ? OnOffType.ON : OnOffType.OFF);
-                }
-                if (basicChannel.getType().equals("Color")) {
-                    Color rgb = new Color(val.getAsInt());
-                    HSBType hsb = HSBType.fromRGB(rgb.getRed(), rgb.getGreen(), rgb.getBlue());
-                    updateState(basicChannel.getChannel(), hsb);
-                }
-            } catch (Exception e) {
-                logger.debug("Error updating {} property {} with '{}' : {}: {}", getThing().getUID(),
-                        basicChannel.getChannel(), val, e.getClass().getCanonicalName(), e.getMessage());
-                logger.trace("Property update error detail:", e);
+            if (basicChannel.getType().equals("String")) {
+                updateState(basicChannel.getChannel(), new StringType(val.getAsString()));
             }
-        } else {
-            logger.debug("Channel not found for {}", param);
+            if (basicChannel.getType().equals("Switch")) {
+                updateState(basicChannel.getChannel(),
+                        val.getAsString().toLowerCase().equals("on") || val.getAsString().toLowerCase().equals("true")
+                                ? OnOffType.ON
+                                : OnOffType.OFF);
+            }
+            if (basicChannel.getType().equals("Color")) {
+                Color rgb = new Color(val.getAsInt());
+                HSBType hsb = HSBType.fromRGB(rgb.getRed(), rgb.getGreen(), rgb.getBlue());
+                updateState(basicChannel.getChannel(), hsb);
+            }
+        } catch (Exception e) {
```

Comment:
Any specific exceptions you are trying to catch? If so please list them instead of catching a generic exception.

Reply:
There were various updating issues (e.g. conversion issues, colors not updating ell etc)
There should be no particular errors cuurently known to happen here

Reply:
leave as is for now then